pkg/services/ngalert/notifier/external_alertmanager.go

```diff
@@ -121,30 +123,62 @@ func (am *externalAlertmanager) ListSilences(ctx context.Context, filter []strin
 	return res.Payload, nil
 }
 
-func (am *externalAlertmanager) GetStatus() apimodels.GettableStatus {
-	return apimodels.GettableStatus{}
+func (am *externalAlertmanager) GetAlerts(ctx context.Context, active, silenced, inhibited bool, filter []string, receiver string) (apimodels.GettableAlerts, error) {
+	params := amalert.NewGetAlertsParamsWithContext(ctx).
```

Comment:
I don't love that context is put in params instead of passed to GetAlerts. It's not idiomatic Go, so as a reader of the code I feel I need to spend time to understand if there's a reason it's put in params instead of passed as the first argument. Is there a reason for doing it like this? 😄

Reply:
Yes, because the client library is... not beautiful

Reply:
Ah I'm following now, that's a shame :(